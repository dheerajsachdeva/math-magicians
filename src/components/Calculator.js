import './styles.css';
import { useState } from 'react';
import Output from './output';
import calculate from '../logic/calculate';
import Quote from './Quote';

const Calculator = () => {
  const [output, setOutput] = useState(0);

  const handleclick = (e) => {
    setOutput((obj) => calculate(obj, e.target.innerText));
  };

  return (
    <>
      <div className="calculator-grid">
        <Output output={output.next || output.total || 0} />
        <button onClick={handleclick} type="button">AC</button>
        <button onClick={handleclick} type="button">+/-</button>
        <button onClick={handleclick} type="button">%</button>
        <button onClick={handleclick} className="btn-orange" type="button">รท</button>
        <button onClick={handleclick} type="button">7</button>
        <button onClick={handleclick} type="button">8</button>
        <button onClick={handleclick} type="button">9</button>
        <button onClick={handleclick} className="btn-orange" type="button">x</button>
        <button onClick={handleclick} type="button">4</button>
        <button onClick={handleclick} type="button">5</button>
        <button onClick={handleclick} type="button">6</button>
        <button onClick={handleclick} className="btn-orange" type="button">-</button>
        <button onClick={handleclick} type="button">1</button>
        <button onClick={handleclick} type="button">2</button>
        <button onClick={handleclick} type="button">3</button>
        <button onClick={handleclick} className="btn-orange" type="button">+</button>
        <button onClick={handleclick} type="button" className="span-two">0</button>
        <button onClick={handleclick} type="button">.</button>
        <button onClick={handleclick} className="btn-orange" type="button">=</button>
      </div>
      <Quote />
    </>
  );
};

export default Calculator;
