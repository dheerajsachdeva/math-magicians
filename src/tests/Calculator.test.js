import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  it('renders Calculator component', () => {
    render(<Calculator />);
    const buttons = screen.getAllByRole('button');
    expect(buttons).toHaveLength(19);
  });

  test('snapshot Calculator component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
