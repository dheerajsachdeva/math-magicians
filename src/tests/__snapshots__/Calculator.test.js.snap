// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="calculator-content"
>
  <div
    className="calculator-heading"
  >
    Let's do some Math!
  </div>
  <div
    className="calculator-grid"
  >
    <div
      className="output"
    >
      0
    </div>
    <button
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="btn-orange"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="btn-orange"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="btn-orange"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="btn-orange"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="span-two"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="btn-orange"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
